tls/extensions/s2n_psk_key_exchange_modes.c

```diff
@@ -38,8 +38,13 @@ const s2n_extension_type s2n_psk_key_exchange_modes_extension = {
 
 static bool s2n_psk_key_exchange_modes_should_send(struct s2n_connection *conn)
 {
-    /* Only send a psk_key_exchange_modes extension if psks available */
-    return conn->psk_params.psk_list.len > 0;
+    /* Send a psk_key_exchange_modes extension whenever we are trying to use psks.
+     * We send the psk_key_exchange_mode in the first flight of messages for
+     * interoperability with other TLS implementations. The final check for
+     * psk_list.len > 0 is necessary because external PSKs are used without
+     * setting `use_tickets` or `use_session_cache` to true.
+     */
```

Comment:
You could include the line from the RFC related to this as a compliance comment. You'd still probably want an comment explaining though, since the ambiguity of that line caused this in the first place.